src/pages/academy/components/video-banner/carousel/_custom-carousel.tsx

```diff
@@ -25,63 +25,96 @@ import { useRecursiveTimeout } from 'components/hooks/use-recursive-timeout'
+
+export type ButtonsProps = HTMLAttributes<HTMLDivElement> & {
+    color?: string
+    enabled?: boolean
+    is_enabled?: boolean
+    left?: boolean
+    is_reviews?: boolean
+    disabled?: boolean
+    style?: CustomBlogVideoNavStyleType | string
+}
+
+type PrevAndNextButtonsProps = Pick<
+    ButtonsProps,
+    'color' | 'enabled' | 'is_reviews' | 'onClick' | 'style'
+>
+
+export const PrevButton = ({
+    color,
+    enabled,
+    is_reviews,
+    onClick,
+    style,
+}: PrevAndNextButtonsProps) => (
     <StyledButtonWrapper
         onClick={onClick}
         disabled={!enabled}
         left
         style={style}
         is_reviews={is_reviews}
     >
-        {color === 'black' ? (
-            <ChevronLeft black="true" />
-        ) : color === 'red' ? (
-            <ChevronLeft red="true" />
-        ) : color === 'custom' ? (
-            <ChevronLeft custom="true" />
-        ) : (
-            <ChevronLeft />
-        )}
+        {color ? <ChevronLeft color={color} /> : <ChevronLeft />}
     </StyledButtonWrapper>
 )
 
-PrevButton.propTypes = {
-    chevron_style: PropTypes.object,
-    enabled: PropTypes.bool,
-    onClick: PropTypes.func,
-}
+type NavigationButtonProps = Pick<ButtonsProps, 'color' | 'is_enabled' | 'onClick'>
 
 // TODO: will remove later,not using this for now
-export const NavigationButton = ({ color, is_enabled, onClick }) => (
+export const NavigationButton = ({ color, is_enabled, onClick }: NavigationButtonProps) => (
     <StyledDot onClick={onClick} color={is_enabled ? color : null} />
 )
 
-NavigationButton.propTypes = {
-    color: PropTypes.string,
-    is_enabled: PropTypes.bool,
-    onClick: PropTypes.func,
-}
-
-export const NextButton = ({ color, enabled, is_reviews, onClick, style }) => (
+export const NextButton = ({
+    color,
+    enabled,
+    is_reviews,
+    onClick,
+    style,
+}: PrevAndNextButtonsProps) => (
     <StyledButtonWrapper
         onClick={onClick}
         disabled={!enabled}
         style={style}
         is_reviews={is_reviews}
     >
-        {color === 'black' ? (
-            <ChevronRight black="true" />
-        ) : color === 'red' ? (
-            <ChevronRight red="true" />
-        ) : color === 'custom' ? (
-            <ChevronRight custom="true" />
-        ) : (
-            <ChevronRight />
-        )}
+        {color ? <ChevronRight color={color} /> : <ChevronRight />}
     </StyledButtonWrapper>
 )
 
-NextButton.propTypes = PrevButton.propTypes
+type ChevronStyleType = {
+    chevron_color: string
+    chevron_left: string
+    chevron_right: string
+    is_displayed_on_mobile: boolean
+}
+type CustomBlogVideoNavStyleType = {
+    custom_chevron_color: string
+    custom_is_displayed_on_mobile: boolean
+}
+type NavigationStyleType = {
+    nav_color: string
+    bottom_offset: string
+}
+
+type CustomCarouselProps = {
+    autoplay_interval: number
+    chevron_style: ChevronStyleType
+    children: React.ReactElement[]
+    container_style: string
```

Comment:
isn't this an object of css properties? Can we use `CSSProperties` here?

Reply:
No in this case because we use for chevron_style snake case properties
<img width="382" alt="Screenshot 2022-08-03 at 17 18 27" src="https://user-images.githubusercontent.com/103182473/182631644-c0ac7b23-474d-4f89-95e9-70626defe0ed.png">